### PR TITLE
refactor: migrate to `cosmos/gogoproto`

### PR DESCRIPTION
**Pull request by julienrbrt**
The SDK is moving away from `gogo/protobuf` to use our fork  `cosmos/gogoproto`. This fork is based on Regen's fork, which we have been using for sometimes (https://github.com/cosmos/cosmos-sdk/blob/main/go.mod#L301).

Currently, we are blocked because we are importing TM types (https://github.com/cosmos/cosmos-sdk/pull/13070#issuecomment-1234363235) and I think it would be a win for both to switch to `cosmos/gogoproto` as `gogo/protobuf` is currently not maintained anymore.

---

#### PR checklist

- [x] Tests written/updated, or no tests needed
- [x] `CHANGELOG_PENDING.md` updated, or no changelog entry needed
- [x] Updated relevant documentation (`docs/`) and code comments, or no
      documentation updates needed

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -174,6 +174,8 @@ proto-gen: check-proto-deps
 	@echo "Generating Protobuf files"
 	@go run github.com/bufbuild/buf/cmd/buf generate
 	@mv ./proto/tendermint/abci/types.pb.go ./abci/types/
+	@cp ./proto/tendermint/rpc/grpc/types.pb.go ./rpc/grpc
+	@go mod tidy
```
Comment (thanethomson) [inline Makefile:178]:
```suggestion
```

**Review by thanethomson — COMMENTED** [review]
Who's going to be committing to maintaining this fork over time?

Also, what differences are there between the fork and https://github.com/gogo/protobuf?

**Comment by tychoish** [comment]
> Who's going to be committing to maintaining this fork over time?
> 
> Also, what differences are there between the fork and https://github.com/gogo/protobuf?

I believe that the SDK team will be maintaining it. It is the case that gogo/protobuf is *not* presently maintained, so it's probably best that we move to something. The tendermint core team has long been of the opinion that Tendermint should move to using the upstream/default protobuf generation rather than _any_ fork, but all other things being equal, the cosmos protobuf generator is probably better than gogo proto, though I'd still prefer to use stock prototobuf.

**Comment by thanethomson** [comment]
Cool, thanks for this @julienrbrt :+1: I see some minor conflicts have come up here, but otherwise this should be good to merge.

**Comment by julienrbrt** [comment]
> > Who's going to be committing to maintaining this fork over time?
> > Also, what differences are there between the fork and https://github.com/gogo/protobuf?
> 
> I believe that the SDK team will be maintaining it. It is the case that gogo/protobuf is _not_ presently maintained, so it's probably best that we move to something. The tendermint core team has long been of the opinion that Tendermint should move to using the upstream/default protobuf generation rather than _any_ fork, but all other things being equal, the cosmos protobuf generator is probably better than gogo proto, though I'd still prefer to use stock prototobuf.

Right, the forks adds as well castrepeated support (https://github.com/gogo/protobuf/pull/658).

**Comment by marbar3778** [comment]
if this is merged in a short period could we include it in 0.37?

**Review by thanethomson — APPROVED** [review]
Seems reasonable to me. Thanks @julienrbrt!

**Comment by marbar3778** [comment]
thank you!!

**Pull request merged**